crates/sui-json-rpc/src/api/indexer.rs

```diff
@@ -92,4 +92,21 @@ pub trait IndexerApi {
         /// The Name of the dynamic field
         name: DynamicFieldName,
     ) -> RpcResult<SuiObjectResponse>;
+
+    /// Return the resolved address given resolver and name
+    #[method(name = "resolveNameServiceAddress")]
+    async fn resolve_name_service_address(
+        &self,
+        /// The name to resolve
+        name: String,
+    ) -> RpcResult<SuiAddress>;
+
+    /// Return the resolved names given address,
+    /// if multiple names are resolved, the first one is the primary name.
+    #[method(name = "resolveNameServiceNames")]
```

Comment:
We either should have an API to fetch the default name, _or_ if this is paginated, a `limit` would basically let you fetch _just_ the preferred name.